src/slash_commands/score.rs

```diff
@@ -0,0 +1,139 @@
+use serenity::client::Context;
+use serenity::model::application::command::CommandOptionType;
+use serenity::model::application::interaction::application_command::ApplicationCommandInteraction;
+use serenity::utils::Colour;
+
+use crate::slash_commands::utils::{get_int_option, get_string_option};
+use crate::slash_commands::{SlashCommand, SlashCommandResult};
+use crate::Chombot;
+
+const HAND_COMMAND: &str = "score";
+const HAN_OPTION: &str = "han";
+const FU_OPTION: &str = "fu";
+const MODE_OPTION: &str = "mode";
+
+const DEFAULT_MODE: &str = "default";
+const LOOSE_MODE: &str = "loose";
+const UNLIMITED_MODE: &str = "unlimited";
+
+const MAX_HAN: i64 = 1600;
+const MAX_FU: i64 = 100000;
+
+pub struct ScoreCommand;
+
+impl ScoreCommand {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+#[async_trait]
+impl SlashCommand for ScoreCommand {
+    fn get_name(&self) -> &'static str {
+        HAND_COMMAND
+    }
+
+    fn add_application_command(&self, command: &mut CreateApplicationCommand) {
+        command
+            .description("Calculate the score for given number of han and fu points")
+            .create_option(|option| {
+                option
+                    .name(HAN_OPTION)
+                    .description("Number of han points")
+                    .kind(CommandOptionType::Integer)
+                    .max_int_value(MAX_HAN)
+                    .required(true)
+            })
+            .create_option(|option| {
+                option
+                    .name(FU_OPTION)
+                    .description("Number of fu points")
+                    .kind(CommandOptionType::Integer)
+                    .max_int_value(MAX_FU)
+                    .required(true)
+            })
+            .create_option(|option| {
+                option
+                    .name(MODE_OPTION)
+                    .description("Calculating mode")
+                    .kind(CommandOptionType::String)
+                    .add_string_choice("Default", DEFAULT_MODE)
+                    .add_string_choice("Loose", LOOSE_MODE)
+                    .add_string_choice("Unlimited", UNLIMITED_MODE)
+                    .required(false)
+            });
+    }
+
+    async fn handle(
+        &self,
+        ctx: &Context,
+        command: &ApplicationCommandInteraction,
+        _chombot: &Chombot,
+    ) -> SlashCommandResult {
+        let han = get_int_option(&command.data.options, HAN_OPTION).ok_or("Missing han value")?;
+        let fu = get_int_option(&command.data.options, FU_OPTION).ok_or("Missing fu value")?;
+        let mode = get_string_option(&command.data.options, MODE_OPTION).unwrap_or(DEFAULT_MODE);
+        let points_calculation_mode = match mode {
+            DEFAULT_MODE => Ok(PointsCalculationMode::Default),
+            LOOSE_MODE => Ok(PointsCalculationMode::Loose),
+            UNLIMITED_MODE => Ok(PointsCalculationMode::Unlimited),
+            _ => Err(format!("Invalid mode: {mode}")),
+        }?;
```

Comment:
nit: mayyyybe we could extract this to an `impl FromStr for PointsCalculationMode` instead? :thinking:

Reply:
We can't :disappointed: Both `FromStr` and `PointsCalculationMode` are coming from external crates, so the Rust compiler won't allow us.

Technically this could be moved to `riichi-hand-rs` library, but I don't think it makes sense - it doesn't add any value there, and forces one specific way of converting strings to enum values.

Reply:
My bad, I constantly forget about this condition :facepalm: and yes, I'd consider these string values to be chombot-specific so moving it there makes no sense.